jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/query/TestQueryHints.java

```diff
@@ -84,10 +73,44 @@ public void testJDBCQueryTimeout() throws Exception {
             em = emf.createEntityManager();
 
             em.getTransaction().begin();
-            Query q = em.createQuery("SELECT x FROM Employee x").setHint(QueryHints.JDBC_TIMEOUT, TestQueryHints.realTimeout);
-            q.getResultList();
+            em.createQuery("SELECT x FROM Employee x")
+                .setHint(QueryHints.QUERY_TIMEOUT, TestQueryHints.propertyTimeout)
+                .getResultList();
+
+            //Convert the timeout set (MILLISECONDS) to what is expected by the JDBC layer (SECONDS)
+            double queryTimeoutSeconds = TestQueryHints.propertyTimeout / 1000d;
+            //if there was a remainder, it should round up
+            if(queryTimeoutSeconds % 1 > 0){
+                queryTimeoutSeconds += 1;
+            }
+
+            Assert.assertEquals((int)queryTimeoutSeconds, TestQueryHints.statementTimeout);
+
+            em.getTransaction().rollback();
```

Comment:
don't you want to rollback in finally? Probably not a big deal in this particular case but I can imagine someone copy & paste the code...

Reply:
Fair point, I'll move that as suggested